packages/babel-plugin-transform-object-rest-spread/test/fixtures/object-rest/symbol/expected.js

```diff
@@ -0,0 +1,14 @@
+var _ref3;
+
+let _ref = {},
+    {
+  [Symbol.for("foo")]: foo
```

Comment:
The indexing expression needs to be hoisted into a temp var to avoid multiple evaluation. It's kinda safe to not do it in this example because `Symbol.for` is supposed to be idempotent, the bracket value can be an arbitrary expression that can have side effects.

You can check if the expression is safe to not hoist by calling `.isStatic()` on the path. It is a bit too conservative and will return false for literal strings and numbers, though; you might want to also check for `isPureish()` (TIL "Pureish" is a thing in babel).